e2e/tests/plugin_metadata.crud-all-fields.spec.ts

```diff
@@ -0,0 +1,185 @@
+
+    // Add Plugin dialog should appear
+    const addPluginDialog = page.getByRole('dialog', { name: 'Add Plugin' });
+    await expect(addPluginDialog).toBeVisible();
+
+    // Fill in comprehensive configuration with all available fields
+    const pluginEditor = await uiGetMonacoEditor(page, addPluginDialog);
+    await uiFillMonacoEditor(
+      page,
+      pluginEditor,
+      JSON.stringify({
+        log_format: {
+          host: '$host',
+          client_ip: '$remote_addr',
+          request_method: '$request_method',
+          request_uri: '$request_uri',
+          status: '$status',
+          body_bytes_sent: '$body_bytes_sent',
+          request_time: '$request_time',
+          upstream_response_time: '$upstream_response_time',
+        },
+      })
+    );
+
+    // Click Add button
+    await addPluginDialog.getByRole('button', { name: 'Add' }).click();
+
+    // Should show success message
+    await uiHasToastMsg(page, {
+      hasText: 'success',
+    });
+
+    // Dialog should close
+    await expect(addPluginDialog).toBeHidden();
+
+    // Plugin card should now be visible
+    const httpLoggerCard = page.getByTestId('plugin-http-logger');
+    await expect(httpLoggerCard).toBeVisible();
+  });
+
+  await test.step('edit plugin metadata with extended fields', async () => {
+    // Find the http-logger card
+    const httpLoggerCard = page.getByTestId('plugin-http-logger');
+
+    // Click Edit button
+    await httpLoggerCard.getByRole('button', { name: 'Edit' }).click();
+
+    // Edit Plugin dialog should appear
+    const editPluginDialog = page.getByRole('dialog', { name: 'Edit Plugin' });
+    await expect(editPluginDialog).toBeVisible();
+
+    // Verify existing configuration is shown
+    await expect(editPluginDialog.getByText('log_format')).toBeVisible();
+
+    // Update the configuration with additional fields
+    const pluginEditor = await uiGetMonacoEditor(page, editPluginDialog);
+    await uiFillMonacoEditor(
+      page,
+      pluginEditor,
+      JSON.stringify({
+        log_format: {
+          host: '$host',
+          client_ip: '$remote_addr',
+          request_method: '$request_method',
+          request_uri: '$request_uri',
+          status: '$status',
+          body_bytes_sent: '$body_bytes_sent',
+          request_time: '$request_time',
+          upstream_response_time: '$upstream_response_time',
+          time: '$time_iso8601',
+          user_agent: '$http_user_agent',
+        },
+      })
+    );
+
+    // Click Save button
+    await editPluginDialog.getByRole('button', { name: 'Save' }).click();
+
+    // Should show success message
+    await uiHasToastMsg(page, {
+      hasText: 'success',
+    });
+
+    // Dialog should close
+    await expect(editPluginDialog).toBeHidden();
+  });
+
+  await test.step('verify configuration changes were saved', async () => {
+    // Verify changes via API
+    const response = await e2eReq.get(`${API_PLUGIN_METADATA}/http-logger`);
```

Comment:
Please do not arbitrarily use APIs during the testing process unless you are preparing resources for testing or cleaning up resources.

Here, you can click the edit button in the UI and then check whether the edited content is displayed on the UI.